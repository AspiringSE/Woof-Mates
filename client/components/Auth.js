import React, { Component } from 'react';
import { connect } from 'react-redux'
import Login from './Login'
import Logout from './Logout'
import { Redirect } from 'react-router-dom'
import { useHistory } from 'react-router-dom';

class Auth extends Component {
    constructor(props) {
        super(props);
        this.state = {
            message: '',
        }
        this.setMessage = this.setMessage.bind(this)
    }
    setMessage(message){
        if (message.includes('Welcome')) {
            this.props.history.push({
                pathname: '/match'
            })
        }
        this.setState({ message })
    }
    render() {
        return (
            <div id="signInContainer">
                {this.props.user.firstName ? <Logout setMessage={this.setMessage} /> : <Login setMessage={this.setMessage} />}
                <p>{this.state.message}</p>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        user: state.user
    }
};

export default connect(mapStateToProps)(Auth);
