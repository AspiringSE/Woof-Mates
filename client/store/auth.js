import axios from 'axios';

const LOGIN = 'LOGIN';

const _login = (user) => {
    return {
        type: LOGIN,
        user
    }
};

const login = (userObj) => {
    return async(dispatch) => {
        try {
            const user = await (axios.get('/api/auth/login', userObj))
            dispatch(_login(user))
        } catch(err) { console.error(err); }
    }
}

export default function userReducer(state = {}, action) {
    switch (action.type) {
        case LOGIN: 
            return action.user;
        default: 
            return state
    }
}

export { login };