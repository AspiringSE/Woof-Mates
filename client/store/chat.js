import axios from 'axios';

const GET_CHAT = 'GET_CHAT';

const setChat = (chat) => {
    return {
        type: GET_CHAT,
        chat
    }
}

export const getChat = (fromId, toId) => async(dispatch) => {
    try {
        const { data } = await axios.get(`api/firebaseChat/${fromId}/${toId}`);
        dispatch(setChat(data))
    } catch (err) {
        console.log(err)
    }
}

const SEND_CHAT = 'SEND_CHAT';

const send_Chat = (message) => {
    return {
        type: SEND_CHAT,
        message
    }
}

export const sendChat = (fromId, toId, message, timestamp, from, to) => async(dispatch) => {
    try {
        await axios.put(`/api/firebasechat/${fromId}/${toId}`, { message, timestamp, from, to});
        dispatch(send_Chat({from, message, timestamp, to}))
    } catch (error) {
        console.log(error)
    }
}

export default function chatReducer (state = [], action) {
    switch (action.type) {
        case GET_CHAT:
            return action.chat
        case SEND_CHAT:
            return [...state, action.message]
        default: 
            return state
    }
}

