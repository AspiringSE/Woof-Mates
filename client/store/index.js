import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { logger } from 'redux-logger';
import userReducer from './user';
import matchReducer from './match';
import matchesReducer from './matches';
import testimonialsReducer from './testimonials'

const reducer = combineReducers({
    user: userReducer,
    match: matchReducer,
    matches: matchesReducer,
    testimonials: testimonialsReducer
});

const store = createStore(reducer, applyMiddleware(logger,thunkMiddleware));

export default store;
