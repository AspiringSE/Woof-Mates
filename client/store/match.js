import axios from 'axios';

const GET_MATCH = 'GET_MATCH';

const _getMatch = (match) => {
    return {
        type: GET_MATCH,
        match
    }
};

const getMatch = (userId) => async(dispatch) => {
    try {
        const { data } = await (axios.get(`/api/match/${userId}`))
        console.log(data)
        dispatch(_getMatch(data))
    } catch (err) { console.error(err); }
};

const sendDecision = (userId, matchId, decision) => async() => {
    try {
        const relationship = (await (axios.put(`/api/match/${userId}`, { decision, matchId }))).data;
        return relationship;
    } catch (err) { console.error(err); }
}

export default function matchReducer(state = {}, action) {
    switch (action.type) {
        case GET_MATCH:
            return action.match
        default:
            return state
    }
}

export { getMatch, sendDecision }
