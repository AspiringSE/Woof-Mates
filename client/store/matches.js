import axios from 'axios';

const GET_MATCHES = 'GET_MATCHES';

const _getMatches = (matches) => {
    return {
        type: GET_MATCHES,
        matches
    }
};

const getMatches = (userId) => async(dispatch) => {
    try {
        //list of all matches returns the 'matchId'. 'matchId' refers to the id that is not the userId from the api route
        const { data } = await (axios.get(`/api/matches/${userId}`));

        //map through above 'matchId' to get the individual user info
        const getAllMatches = await Promise.all(data.map(matchedId => axios.get(`api/users/${matchedId}`)));
        
        //map through to get user data for each
        const allMatches = getAllMatches.map(match => {
            return match.data
        });

        dispatch(_getMatches(allMatches));

    } catch(err) { console.error(err); };
};

export default function matchesReducer(state = [], action) {
    switch (action.type) {
        case GET_MATCHES:
            return action.matches
        default:
            return state
    }
};

export { getMatches };