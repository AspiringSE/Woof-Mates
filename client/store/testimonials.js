import axios from 'axios';

const GET_TESTIMONIALS = 'GET_TESTIMONIALS';
const POST_TESTIMONIAL = 'POST_TESTIMONIAL'

const _getTestimonials = (testimonials) => {
  return {
      type: GET_TESTIMONIALS,
      testimonials
  }
};

const _postTestimonial = (testimonial) => {
  return {
    type: POST_TESTIMONIAL,
    testimonial
  }
}

export const getTestimonials = () => {
  return async(dispatch) => {
    try {
      const { data } = await axios.get('/api/testimonials')
      dispatch(_getTestimonials(data));
    } catch (err) {console.error(err)}
  }
};

export const postTestimonial = (reviewInfo) => {
  return async(dispatch) => {
    try {
      const newTestimonial = (await axios.post('/api/testimonials', reviewInfo)).data
      dispatch(_postTestimonial(newTestimonial))
    } catch (err) {console.error(err)}
  }
}

export default function testimonialsReducer(state = [], action) {
    switch (action.type) {
        case GET_TESTIMONIALS:
          return action.testimonials
        case POST_TESTIMONIAL:
          return [...state, action.testimonial]
        default:
          return state
    }
}
