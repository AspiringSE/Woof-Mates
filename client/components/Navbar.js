import React from 'react';
import { connect } from 'react-redux'
import { HashRouter as Router, Link, Switch, Route }
from 'react-router-dom';
import { logout } from '../store/user'


class Navbar extends React.Component {
  constructor(props) {
    super(props)
    this.logout = this.logout.bind(this)
    this.logIn = this.logIn.bind(this)
    this.signUp = this.signUp.bind(this)
  }

  componentDidMount () {
    const {user} = this.props
  }

  logout (e) {
    e.preventDefault();
    this.props.logout(this.props.user.id)
    window.location.hash = '/login'
  }

  logIn (e) {
    e.preventDefault();
    window.location.hash = '/login'
  }

  signUp (e) {
    e.preventDefault();
    window.location.hash = '/signUp'
  }

  render() {
    const {user} = this.props
    if (user.firstName) {
      return (
        <div id="NavbarContainer">
          <div id="NavbarTitleContainer">
            <Link id="NavbarTitleLink" to="/">Woofmates</Link>
          </div>
          <div id="NavbarLogoContainer">
            <img id="NavbarLogoPic" src="https://cdn.shopify.com/s/files/1/0667/0685/files/tintin_and_snowy_by_genus_rattus-d4jxohy_large.png?9525775607535621580"></img>
          </div>
          <div id="LinkContainer">
              <Link id="NavbarLink" to="/">Home</Link>
          </div>
          <div id="LinkContainer">
              <Link id="NavbarLink" to="/profile">My Profile</Link>
          </div>
          <div id="LinkContainer">
              <Link id="NavbarLink" to="/chat">Chat</Link>
          </div>
          <div id="LinkContainer">
              <Link id="NavbarLink" to="/match">Find Your Match</Link>
          </div>
          <button className = "logOutButton" onClick={this.logout}>Log Out</button>
        </div>
      );
    }

    else {
      return (
        <div id="NavbarContainer">
          <div id="NavbarTitleContainer">
            <Link id="NavbarTitleLink" to="/">Woofmates</Link>
          </div>
          <div id="NavbarLogoContainer">
            <img id="NavbarLogoPic" src="https://cdn.shopify.com/s/files/1/0667/0685/files/tintin_and_snowy_by_genus_rattus-d4jxohy_large.png?9525775607535621580"></img>
          </div>
          <div id="LinkContainer">
              <Link id="NavbarLink" to="/">Home</Link>
          </div>
          <div id="LinkContainer">
              <Link id="NavbarLink" to="/profile">My Profile</Link>
          </div>
          <div id="LinkContainer">
              <Link id="NavbarLink" to="/chat">Chat</Link>
          </div>
          <div id="LinkContainer">
              <Link id="NavbarLink" to="/match">Find Your Match</Link>
          </div>
          <div id="LinkContainer">
            <button className="signInNavButton" onClick={this.logIn}>Sign In </button>
          </div>
          <div id="LinkContainer">
            <button className = "signUpNavButton" onClick={this.signUp}>Sign Up</button>
          </div>
        </div>
      )
    }
  }
}

const mapState = state => ({
    user: state.user
})

const mapDispatchToProps = (dispatch) => ({
      logout: (userId) => dispatch(logout(userId))
})

export default connect(mapState, mapDispatchToProps)(Navbar);
