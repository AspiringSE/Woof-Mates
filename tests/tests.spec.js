import React from 'react';
import renderer from 'react-test-renderer';
import Auth from '../client/components/Auth'
import Navbar from '../client/components/Navbar'
import App from '../client/app'
import { Provider } from 'react-redux'
import store from '../client/store'
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router-dom'

Enzyme.configure({ adapter: new Adapter() });

// describe('Examining the syntax of Jest tests', () => {
//     it('sums numbers', () => {
//         expect(1 + 2).toEqual(3);
//         expect(2 + 2).toEqual(4);
//      });
//   });
describe('Login tests', () => {
  test('Auth component renders correctly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Auth />
      </Provider>
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  test('Clicking navbar sign up button renders login page', () => {
    let app = mount(
      <Provider store={store}>
        <MemoryRouter >
          <App />
        </MemoryRouter>
      </Provider>
      )
    app.find('.signInNavButton').at(1).simulate('click', { button: 0 })
    app.update();
    expect(app.text().includes('Welcome!')).toBe(true);
    // expect(app.find('#signInContainer')).toHaveLength(1)
    
  })
})
