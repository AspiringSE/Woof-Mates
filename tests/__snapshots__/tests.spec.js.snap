// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth component renders correctly 1`] = `
<div
  id="signInContainer"
>
  <div
    id="signInContainer"
  >
    <div
      id="signInForm"
    >
      <h3>
        Welcome!
      </h3>
      <form
        id="signInForm"
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiTextField-root loginEmail MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled Mui-required Mui-required"
            data-shrink={false}
            htmlFor="login-email"
            id="login-email-label"
          >
            Your Email
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="email"
              autoFocus={false}
              className="MuiInputBase-input MuiFilledInput-input"
              disabled={false}
              id="login-email"
              name="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
        </div>
        <p />
        <div
          className="MuiFormControl-root MuiTextField-root loginPassword MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled Mui-required Mui-required"
            data-shrink={false}
            htmlFor="login-password"
            id="login-password-label"
          >
            Password
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="current-password"
              autoFocus={false}
              className="MuiInputBase-input MuiFilledInput-input"
              disabled={false}
              id="login-password"
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="password"
              value=""
            />
          </div>
        </div>
        <p />
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained signInAuthButton MuiButton-containedSecondary MuiButton-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Log In
          </span>
        </button>
      </form>
    </div>
  </div>
  <p>
    
  </p>
</div>
`;

exports[`Login tests Auth component renders correctly 1`] = `
<div
  id="signInContainer"
>
  <div
    id="signInContainer"
  >
    <div
      id="signInForm"
    >
      <h3>
        Welcome!
      </h3>
      <form
        id="signInForm"
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiTextField-root loginEmail MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled Mui-required Mui-required"
            data-shrink={false}
            htmlFor="login-email"
            id="login-email-label"
          >
            Your Email
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="email"
              autoFocus={false}
              className="MuiInputBase-input MuiFilledInput-input"
              disabled={false}
              id="login-email"
              name="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
        </div>
        <p />
        <div
          className="MuiFormControl-root MuiTextField-root loginPassword MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled Mui-required Mui-required"
            data-shrink={false}
            htmlFor="login-password"
            id="login-password-label"
          >
            Password
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="current-password"
              autoFocus={false}
              className="MuiInputBase-input MuiFilledInput-input"
              disabled={false}
              id="login-password"
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="password"
              value=""
            />
          </div>
        </div>
        <p />
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained signInAuthButton MuiButton-containedSecondary MuiButton-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Log In
          </span>
        </button>
      </form>
    </div>
  </div>
  <p>
    
  </p>
</div>
`;
